temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingSpanUtils.java

```diff
@@ -0,0 +1,150 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import com.google.common.base.MoreObjects;
+import io.opentracing.References;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+public class OpenTracingSpanUtils {
+  public static Tracer.SpanBuilder createWorkflowStartSpan(
+      Tracer tracer, long startTimeMs, String spanOperationName, String workflowId) {
+    Map<String, String> tags =
+        new HashMap<String, String>() {
+          {
+            put("workflowTag", workflowId);
+          }
+        };
```

Comment:
You can use ImmutableMap.of instead.

Reply:
Yeah, I thought about it, but I actually don't bring a dependency on guava in this module. I know that `temporal-sdk` has a guava dependency anyway, but I don't see enough reasons to make this module dependant on guava too JUST for this small helper inline map factory method. I will maybe consider if at least something more useful from there will be needed.

Reply:
We have an API dependency on temporal-sdk, which should bring in guava.

Reply:
That's exactly what I'm saying. You don't suppose to use transitive dependencies in your module, it's actually not a good style. And this is why Gradle has a separation between implementation and API scope. And, by the way, guava is defined as implementation scoped (as it should be). Whenever you use a library directly in your module you should have an own direct dependency defined in the project module. And I don't want to bring the whole guava for a small factory method, it's not a good practice.

Reply:
Agree with @Spikhalskiy on not bringing large direct dependencies without a good reason.

Reply:
Another suggestion brought guava anyway, so now replaced with ImmutableMap